src/main/java/org/truffleruby/language/methods/Arity.java

```diff
@@ -42,20 +43,29 @@ public Arity(
             boolean hasRest,
             int postRequired,
             String[] keywordArguments,
+            boolean isAllKeywordsOptional,
             boolean hasKeywordsRest) {
         this.preRequired = preRequired;
         this.optional = optional;
         this.hasRest = hasRest;
         this.postRequired = postRequired;
         this.keywordArguments = keywordArguments;
+        this.isAllKeywordsOptional = isAllKeywordsOptional;
```

Comment:
Actually you don't need to store it as a field, just pass it like `computeArityNumber(isAllKeywordsOptional)` 2 lines below :)

Nitpick: I'd name it `allKeywordsOptional` (or `areAllKeywordsOptional` but it doesn't read so nicely).

Reply:
We have `Arity#withRest and `Arity#consumingFirstRequired` methods with returning the new Arity instance based on the current. So have to keep it I guess
Btw, agree with the `allKeywordsOptional` :+1:

Reply:
Indeed, should have noticed that, sorry.